perfkitbenchmarker/benchmarks/iperf_benchmark.py

```diff
@@ -71,32 +76,56 @@ def _RunIperf(sending_vm, receiving_vm, receiving_ip_address, ip_type):
     ip_type: The IP type of 'ip_address' (e.g. 'internal', 'external')
   Returns:
     A Sample.
-  Raises:
-    regex_util.NoMatchError: When iperf results are not found in stdout.
   """
-  iperf_cmd = ('iperf --client %s --port %s --format m --time 60' %
-               (receiving_ip_address, IPERF_PORT))
+  iperf_cmd = ('iperf --client %s --port %s --format m --time %s -P %s' %
+               (receiving_ip_address, IPERF_PORT,
+                FLAGS.iperf_runtime_in_seconds,
+                FLAGS.iperf_sending_thread_count))
   stdout, _ = sending_vm.RemoteCommand(iperf_cmd, should_log=True)
 
-  iperf_pattern = re.compile(r'(\d+\.\d+|\d+) Mbits/sec')
-  value = regex_util.ExtractFloat(iperf_pattern, stdout)
+  # Example output from iperf that needs to be parsed
+  # STDOUT: ------------------------------------------------------------
+  # Client connecting to 10.237.229.201, TCP port 5001
+  # TCP window size: 0.04 MByte (default)
+  # ------------------------------------------------------------
+  # [  6] local 10.76.234.115 port 53527 connected with 10.237.229.201 port 5001
+  # [  3] local 10.76.234.115 port 53524 connected with 10.237.229.201 port 5001
+  # [  4] local 10.76.234.115 port 53525 connected with 10.237.229.201 port 5001
+  # [  5] local 10.76.234.115 port 53526 connected with 10.237.229.201 port 5001
+  # [ ID] Interval       Transfer     Bandwidth
+  # [  4]  0.0-60.0 sec  3730 MBytes  521.1 Mbits/sec
+  # [  5]  0.0-60.0 sec  3499 MBytes   489 Mbits/sec
+  # [  6]  0.0-60.0 sec  3044 MBytes   425 Mbits/sec
+  # [  3]  0.0-60.0 sec  3738 MBytes   522 Mbits/sec
+  # [SUM]  0.0-60.0 sec  14010 MBytes  1957 Mbits/sec
+
+  thread_values = re.findall('\[SUM].*\s+(\d+\.?\d*).Mbits/sec', stdout)
+  if (thread_values is None) or (len(thread_values) == 0):
```

Comment:
Nit: this can just be `if not thread_values:`

Both `None` and `[]` are falsey.